test/index.js

```diff
@@ -502,45 +516,87 @@ function itBehavesLikeWhenLimitdIsRunning(options) {
+              event: 'onRequest'
+            }
+          ], done);
+        });
 
-      after(server.stop);
+        after(server.stop);
 
-      it('should send response with 429 if limit has passed for some plugin configuration and set limit header', function(done){
-        const request = { method: 'POST', url: '/users', payload: { } };
-        server.inject(request, res => {
-          const body = JSON.parse(res.payload);
-          const headers = res.headers;
+        it('should send response with 429 if limit has passed for some plugin configuration and set limit header', function(done){
+          const request = { method: 'POST', url: '/users', payload: { } };
+          server.inject(request, res => {
+            const body = JSON.parse(res.payload);
+            const headers = res.headers;
 
-          expect(res.statusCode).to.equal(429);
-          expect(body.error).to.equal('Too Many Requests');
+            expect(res.statusCode).to.equal(429);
+            expect(body.error).to.equal('Too Many Requests');
 
-          expect(headers['x-ratelimit-limit']).to.equal(0);
-          expect(headers['x-ratelimit-remaining']).to.equal(0);
-          expect(headers['x-ratelimit-reset']).to.equal(0);
+            expect(headers['x-ratelimit-limit']).to.equal(0);
+            expect(headers['x-ratelimit-remaining']).to.equal(0);
+            expect(headers['x-ratelimit-reset']).to.equal(0);
 
-          done();
+            done();
+          });
+        });
+      });
+
+      describe('and when limitResponseHandler is provided', () => {
+        before((done) => {
+          server.start({ replyError: false }, [
+            {
+              type: options.bucket3type,
+              limitd: getLimitdClient(address),
+              extractKey: (request, reply, done) => { done(null, 'key'); },
+              event: 'onRequest',
+              limitResponseHandler: (result, request, reply) => reply.continue()
+            },
+            {
+              type: options.emptyType,
+              limitd: getLimitdClient(address),
+              extractKey: (request, reply, done) => { done(null, 'key'); },
+              event: 'onRequest',
+              limitResponseHandler: (result, request, reply) => reply.continue()
+            },
+            {
+              type: options.bucket3type,
+              limitd: getLimitdClient(address),
+              extractKey: (request, reply, done) => { done(null, 'key'); },
+              event: 'onRequest',
+              limitResponseHandler: (result, request, reply) => reply.continue()
+            }
+          ], done);
+        });
+
+        after(server.stop);
+
+        it('should call limitResponseHandler to handle the answer', function(done){
+          const request = { method: 'POST', url: '/users', payload: { } };
+          server.inject(request, res => {
+            expect(res.statusCode).to.equal(200);
```

Comment:
Would it make sense to spy on `limitResponseHandler` and check that it was actually invoked, to eliminate the possibility that the `take` returned a conformant response? Otherwise, I think that configuring only the `options.emptyType` bucket would make it more obvious that `take` returns a non-conformant result.

---

I'd also add a test with a custom handler that returns non-429 result (to witness the handler overriding the default error code). That would also self-document one use-case showing how to use the new handler.

Reply:
I don't think that the kind of test patova has <that are more integration tests> let us spy in an easy way and non-hacky way. Also we have other test above for the same condition that check that there is a 429 response in this case. 

I could maybe check the response handler for the result if we want to be 100% sure in this particular test case